rotkehlchen/accounting/cost_basis/base.py

```diff
@@ -101,10 +104,15 @@ def __str__(self) -> str:
 
 
 class BaseAcquisitionsOrder(metaclass=ABCMeta):
-    _acquisitions: deque[AssetAcquisitionEvent]
+    # For FIFO, the time at the point of insertion is used
+    # For LIFO, the time at the point of insertion is used with a negation
+    # For HIFO, the amount of the acquisition event is used with a negation
+    # The reason for the negation is because of Python's `heapq` implementation
+    Priority = Union[FVal, Timestamp]
+    _acquisitions: list[Tuple[Priority, AssetAcquisitionEvent]]
 
     def __init__(self) -> None:
-        self._acquisitions = deque()
+        self._acquisitions = []
```

Comment:
Maybe we should state in a comment that this is supposed to be a priority queue (to prevent someone accessing/modifying it directly)

Reply:
absolutely. This is one of the most common bugs in python heapq usage and why many people don't use that hacky library.

It offers no protection of using the underlying list again at any point. It's a terrible api. So easy to mess up.